lib/src/benchmark_base.dart

```diff
@@ -62,3 +62,64 @@ class BenchmarkBase {
     emitter.emit(name, measure());
   }
 }
+
+class AsyncBenchmarkBase {
+  final String name;
+  final ScoreEmitter emitter;
+
+  // Empty constructor.
+  const AsyncBenchmarkBase(this.name, {this.emitter = const PrintEmitter()});
+
+  // The benchmark code.
+  // This function is not used, if both [warmup] and [exercise] are overwritten.
+  run() async {}
+
+  // Runs a short version of the benchmark. By default invokes [run] once.
+  void warmup() async {
+    await run();
+  }
+
+  // Exercices the benchmark. By default invokes [run] 10 times.
+  void exercise() async {
+    for (int i = 0; i < 10; i++) {
+      await run();
+    }
+  }
+
+  // Not measured setup code executed prior to the benchmark runs.
+  setup() async {}
+
+  // Not measures teardown code executed after the benchark runs.
+  teardown() async {}
+
+  // Measures the score for this benchmark by executing it repeately until
+  // time minimum has been reached.
+  static Future<double> measureFor(Function f, int minimumMillis) async {
+    int minimumMicros = minimumMillis * 1000;
```

Comment:
Use final for variables that are assigned once.

Reply:
Done